Mage/src/main/java/mage/abilities/effects/common/PutTokenOntoBattlefieldCopyTargetEffect.java

```diff
@@ -162,18 +179,27 @@ public boolean apply(Game game, Ability source) {
             return false;
         }
 
-        EmptyToken token = new EmptyToken();
+        EmptyToken token = new EmptyToken();        
         CardUtil.copyTo(token).from(copyFrom); // needed so that entersBattlefied triggered abilities see the attributes (e.g. Master Biomancer)
         applier.apply(game, token);
+        
         if (additionalCardType != null && !token.getCardType().contains(additionalCardType)) {
             token.getCardType().add(additionalCardType);
         }
         if (gainsHaste) {
             token.addAbility(HasteAbility.getInstance());
         }
+        if (gainsFlying) {
+            token.addAbility(FlyingAbility.getInstance());
+        }
+        if (tokenPower != 0 || tokenToughness != 0) {
```

Comment:
I think this should use -1 or INT_MIN as the special value, since this design prevents implementing a future card that makes a 0/1 token, for instance.

Also, I think there should be separate if checks and sets for power and toughness so that future cards that only override power or toughness can also be implemented.